### PR TITLE
[Improvement](runtime-filter) use shared ptr to instead object pool to store runtime filters

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
use shared ptr to instead object pool to store runtime filters

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run p0

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/exprs/minmax_predicate.h [hunk 1]
```diff
@@ -75,19 +76,22 @@ class MinMaxNumFunc : public MinMaxFuncBase {
         for (size_t i = start; i < size; i++) {
             if (nullmap == nullptr || !nullmap[i]) {
                 if constexpr (NeedMin) {
-                    _min = std::min(_min, column_string.get_data_at(i));
+                    if (column_string.get_data_at(i) < StringRef(_min)) {
+                        _min = column_string.get_data_at(i).to_string();
+                    }
```
Comment (yiguolei) [inline be/src/exprs/minmax_predicate.h:81]: what about column_string.get_data_at(i) >= StringRef(_min

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Review by Mryange — APPROVED** [review]

**Pull request merged**